src/easylink/configuration.py

```diff
@@ -162,72 +155,27 @@ def spark_resources(self) -> Dict[str, Any]:
             **{k: v for k, v in self.spark.items() if k != "workers"},
         }
 
-    def get_implementation_name(self, step_name: str) -> str:
-        return self.pipeline[step_name].implementation.name
-
     #################
     # Setup Methods #
     #################
-    @staticmethod
-    def _spark_is_required(pipeline: Dict[str, Any]) -> bool:
-        """Check if the pipeline requires spark resources."""
-        implementation_metadata = load_yaml(paths.IMPLEMENTATION_METADATA)
-        try:
-            implementations = [step["implementation"]["name"] for step in pipeline.values()]
-        except Exception as e:
-            raise KeyError(
-                "The pipeline specification should contain keys for each step "
-                "and each step should contain an 'implementation' key with a 'name' key."
-            ) from e
-        for implementation in implementations:
-            if implementation_metadata[implementation].get("requires_spark", False):
-                return True
-        return False
+    def update_implementation_configs(self, level: LayeredConfigTree):
```

Comment:
we can't make too many assumptions about what level of the pipeline config *should* have an implementation, so I update this default recursively.